### PR TITLE
protocols/mdns: Update to if-watch v2.0.0

### LINKED ISSUE #2895: Update to if-watch v2.0.0 in mdns
## Description

As it was done in https://github.com/libp2p/rust-libp2p/pull/2813

## Motivation

```
$ fd Cargo.toml --exec rg -H if-watch {}
protocols/mdns/Cargo.toml:if-watch = "1.1.1"
transports/tcp/Cargo.toml:if-watch = "2.0.0"
```

At least it would be great to compile less code.

### PR DESCRIPTION
**Pull request by umgefahren**
# Description

As suggested I updated the version of if-watch. This should reduce compile time, since the versions used in tcp and mdns are now aligned.

## Links to any relevant issues

Resolves #2895.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by umgefahren — COMMENTED** [review]
Looks good

**Comment by umgefahren** [comment]
Since @mxinden authored if-watch, it would be great if he would take a look :)

File: protocols/mdns/src/behaviour.rs [hunk 1]
```diff
@@ -83,7 +82,7 @@ where
 {
     /// Builds a new `Mdns` behaviour.
     pub async fn new(config: MdnsConfig) -> io::Result<Self> {
```
Comment (thomaseizinger) [inline protocols/mdns/src/behaviour.rs]: This can now be a non-async fn! :tada:

**Review by thomaseizinger — COMMENTED** [review]
One comment, otherwise LGTM!

Thanks!

File: protocols/mdns/CHANGELOG.md [hunk 2]
```diff
@@ -10,9 +10,12 @@
 
 - Removed the `lazy_static` dependency. See [PR 2977].
 
+- Update to `if-watch` `v2.0.0`. See [PR 2978].
```
Comment (mxinden) [inline protocols/mdns/CHANGELOG.md]:
```suggestion
- Update to `if-watch` `v2.0.0` and thus the `async` method `Mdns::new` and `TokioMdns::new` becomes synchronous. See [PR 2978].
```

Might be helpful for people upgrading, though I guess the compile time error will help more.

**Review by mxinden — COMMENTED** [review]
> Since @mxinden authored if-watch, it would be great if he would take a look :)

Actually credit does not go to me but to @dvc94ch. I am only maintaining it.

This looks good to me. One small suggestion.

**Review by mxinden — APPROVED** [review]
:pray:

**Pull request merged**